### PR TITLE
First pass to rst-ify README.tasks.  Still needs cross-references

### PR DESCRIPTION
**Pull request by noakesmichael**

### CODE REVIEW

**Comment by ronawho** [comment]
FYI: using http://web/~eronagha/mike/tasks.html to review

File: doc/release/README.tasks.rst [hunk 1]
```diff
@@ -2,6 +2,12 @@
 Chapel programs, tasks, and threads
 ===================================
 
+.. contents::
```
Comment (ronawho) [inline doc/release/README.tasks.rst]: Can you remove this? I think it adds clutter, and is mostly served by the index on the left hand side

File: doc/release/README.tasks.rst [hunk 2]
```diff
@@ -2,6 +2,12 @@
 Chapel programs, tasks, and threads
```
Comment (ronawho) [inline doc/release/README.tasks.rst]: Mind changing this to just "Chapel Tasks"?

**Comment by ronawho** [comment]
What do you think of removing  "The implementation of tasks and threads depends on the target platform ($CHPL_TARGET_PLATFORM) and tasking package ($CHPL_TASKS) selected by the user. For an introduction to these variables, refer to README.chplenv."?

I don't find it especially helpful, and it's bound to be incomplete since how we choose tasking layers changes over time and relies on things other than target platform and chpl_tasks. Do you think it's useful at all?

File: doc/release/README.tasks.rst (hunk 2 above)
Comment (noakesmichael) [inline doc/release/README.tasks.rst]: Done.

On Sep 21, 2015, at 11:16 AM, Elliot Ronaghan <notifications@github.com<mailto:notifications@github.com>> wrote:


In doc/release/README.tasks.rst<https://github.com/chapel-lang/chapel/pull/2593#discussion_r40005462>:

> @@ -2,6 +2,12 @@
>  Chapel programs, tasks, and threads


Mind changing this to just "Chapel Tasks"?

File: doc/release/README.tasks.rst [hunk 3]
```diff
@@ -90,16 +108,18 @@ qualitatively speaking it will have a greater effect on programs which
 create more or shorter-lived tasks than on programs which create fewer
 or longer-lived ones.
 
-As described below in "Task Call Stacks", the execution-time default for
+As described in `Task Call Stacks`_, the execution-time default for
 stack overflow checking can be set by using the "--[no-]stack-checks"
```
Comment (ronawho) [inline doc/release/README.tasks.rst]: Do you think we should use code blocks for "--[no-]stack-checks"? I would lean towards yes, but don't feel strongly about it.

**Comment by noakesmichael** [comment]
I think you’re right.

Mike

On Sep 21, 2015, at 11:19 AM, Elliot Ronaghan <notifications@github.com<mailto:notifications@github.com>> wrote:


What do you think of removing "The implementation of tasks and threads depends on the target platform ($CHPL_TARGET_PLATFORM) and tasking package ($CHPL_TASKS) selected by the user. For an introduction to these variables, refer to README.chplenv."?

I don't find it especially helpful, and it's bound to be incomplete since how we choose tasking layers changes over time and relies on things other than target platform and chpl_tasks. Do you think it's useful at all?

**Pull request merged**